File: examples/nextjs/src/ledger.ts
import { createServerClient, type ServerClientOptions } from "@bowlinedev/client/server";
import { headers } from "next/headers";
import { type Client, createClient } from "./bowline";

export const backend = `${process.env.LEDGER_URL ?? "http://localhost:8080"}/api`;

export async function ledger(options: Partial<ServerClientOptions> = {}): Promise<Client> {
  const incoming = await headers();
  return createServerClient(createClient, {
    url: backend,
    request: { headers: incoming },
    ...options,
  });
}

File: examples/federation/billing/api/ledger.go
package api

import (
	"net/http"

	"github.com/bowlinedev/bowline/examples/ledger/ledgerclient"
	"github.com/bowlinedev/bowline/signing"
)

func LedgerClient(url, keyID string, secret []byte) *ledgerclient.Client {
	if keyID == "" || len(secret) == 0 {
		return ledgerclient.New(url)
	}
	return ledgerclient.New(url, ledgerclient.WithHTTPClient(&http.Client{
		Transport: &signing.Transport{KeyID: keyID, Secret: secret},
	}))
}
